### DISCUSSION TITLE
error on open some pdf

### DISCUSSION THREAD

**Discussion by robtomasi** [UglyToad/PdfPig#456] [Q&A]
i have this error on open some pdf, on this statement
        foreach (var page in **document.GetPages(**)) of class OpenDocumentAndExtractWords in the   public static string Run(string filePath)
 any ideas?

thanks a lot

UglyToad.PdfPig.Core.PdfDocumentFormatException
  HResult=0x80131500
  Messaggio=**The filter for the stream was not a valid object. Expected name or array, instead got: 18 0.**
  Origine=UglyToad.PdfPig
  Analisi dello stack:
   in UglyToad.PdfPig.Filters.DefaultFilterProvider.GetFilters(DictionaryToken dictionary)
   in UglyToad.PdfPig.PdfExtensions.Decode(StreamToken stream, IFilterProvider filterProvider)
   in UglyToad.PdfPig.Parser.PageFactory.Create(Int32 number, DictionaryToken dictionary, PageTreeMembers pageTreeMembers, Boolean clipPaths)
   in UglyToad.PdfPig.Content.Pages.GetPage(Int32 pageNumber, Boolean clipPaths)
   in UglyToad.PdfPig.PdfDocument.GetPage(Int32 pageNumber)
   in UglyToad.PdfPig.PdfDocument.<GetPages>d__32.MoveNext()
   in ProvaLeggiPdfC.OpenDocumentAndExtractWords.Run(String filePath) in C:\lavoro\Vs2019\ProvaLeggiPdfC\ProvaLeggiPdfC\PingExample\OpenDocumentAndExtractWords.cs: riga 19
   in ProvaLeggiPdfC.frmAutostrade.button4_Click(Object sender, EventArgs e) in C:\lavoro\Vs2019\ProvaLeggiPdfC\ProvaLeggiPdfC\frmAutostrade.cs: riga 40
   in System.Windows.Forms.Control.OnClick(EventArgs e)
   in System.Windows.Forms.Button.OnClick(EventArgs e)
   in System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   in System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   in System.Windows.Forms.Control.WndProc(Message& m)
   in System.Windows.Forms.ButtonBase.WndProc(Message& m)
   in System.Windows.Forms.Button.WndProc(Message& m)
   in System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   in System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   in System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   in System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   in System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   in System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   in System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   in System.Windows.Forms.Application.Run(Form mainForm)
   in ProvaLeggiPdfC.Program.Main() in C:\lavoro\Vs2019\ProvaLeggiPdfC\ProvaLeggiPdfC\Program.cs: riga 19

  Questa eccezione è stata generata in origine nello stack di chiamate seguente:
    [Codice esterno]
    ProvaLeggiPdfC.OpenDocumentAndExtractWords.Run(string) in OpenDocumentAndExtractWords.cs
    ProvaLeggiPdfC.frmAutostrade.button4_Click(object, System.EventArgs) in frmAutostrade.cs
    [Codice esterno]
    ProvaLeggiPdfC.Program.Main() in Program.cs

**Comment by EliotJones** [comment]
This looks like it was raised as issue #457
